core/kernel/src/main/java/io/thorntail/tracing/impl/TracingInterceptor.java

```diff
@@ -82,10 +75,23 @@ protected boolean isTraced(Object target, Method method) {
         return classTraced.value();
     }
 
+    private List<SpanHandler> spanHandlers() {
+        List<SpanHandler> handlers = spanHandlers;
```

Comment:
Why? Is this some smart way to make the number of volatile loads/stores lower?

Reply:
Yes, the local copy ensures we only do 1 volatile read if the cached value is already initialized. It might look like a premature optimization here... but I think it's ok ;-)

Reply:
Ah I see, the `return` statement would have to do another volatile read. I see, thanks.